includes/class-amp-theme-support.php

```diff
@@ -2495,6 +2503,137 @@ public static function enqueue_assets() {
 		wp_styles()->add_data( 'amp-default', 'rtl', 'replace' );
 	}
 
+	/**
+	 * Setup pages to have the paired browsing client script so that the app can interact with it.
+	 *
+	 * @return void
+	 */
+	public static function setup_paired_browsing_client() {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( isset( $_GET[ self::PAIRED_BROWSING_QUERY_VAR ] ) ) {
+			return;
+		}
+
+		$asset_file   = AMP__DIR__ . '/assets/js/amp-paired-browsing-client.asset.php';
+		$asset        = require $asset_file;
+		$dependencies = $asset['dependencies'];
+		$version      = $asset['version'];
+
+		wp_enqueue_script(
+			'amp-paired-browsing-client',
+			amp_get_asset_url( '/js/amp-paired-browsing-client.js' ),
+			$dependencies,
+			$version,
+			true
+		);
+
+		// Force dev mode to be enabled. This ensures that the enqueued script and its dependencies
+		// will be present when the admin bar is not showing.
+		add_filter( 'amp_dev_mode_enabled', '__return_true' );
+
+		// Whitelist enqueued script so that it is not removed.
+		add_filter( 'script_loader_tag', [ __CLASS__, 'filter_paired_browsing_client_script_loader_tag' ], 10, 2 );
+	}
+
+	/**
+	 * Add the `data-ampdevmode` attribute to any enqueued script that the paired browsing interface
+	 * depends on.
+	 *
+	 * @since 1.3
+	 *
+	 * @param string $tag    The link tag for the enqueued script.
+	 * @param string $handle The script's registered handle.
+	 * @return string Tag.
+	 */
+	public static function filter_paired_browsing_client_script_loader_tag( $tag, $handle ) {
+		if ( self::has_dependency( wp_scripts(), 'amp-paired-browsing-client', $handle ) ) {
+			$tag = preg_replace( '/(?<=<script)(?=\s|>)/i', ' ' . AMP_Rule_Spec::DEV_MODE_ATTRIBUTE, $tag );
+		}
+
+		return $tag;
+	}
```

Comment:
FYI: This should be able to be largely eliminated if we implement this: https://github.com/google/site-kit-wp/pull/505#discussion_r348683617